### PR TITLE
Dictionary to store event handlers: updated text for a new example

### PR DESCRIPTION
**Pull request by pkulikov**
This PR originates from the following discussion: https://github.com/dotnet/samples/pull/701#issuecomment-471547444

Should be merged only after the corresponding samples PR is merged.
Supported by dotnet/samples#701

@BillWagner please review.

### CODE REVIEW

**Comment by BillWagner** [comment]
I like the changes, and I'm ready to approve this. 

I'm waiting for the resolution of https://github.com/dotnet/samples/pull/701#issuecomment-472032112

Then, I'll merge both PRs.

File: docs/csharp/programming-guide/events/how-to-use-a-dictionary-to-store-event-instances.md [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 ---
-# How to: Use a Dictionary to Store Event Instances (C# Programming Guide)
-One use for `accessor-declarations` is to expose many events without allocating a field for each event, but instead using a Dictionary to store the event instances. This is only useful if you have many events, but you expect most of the events will not be implemented.  
-  
-## Example  
- [!code-csharp[csProgGuideEvents#9](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs#9)]  
-  
+# How to: use a dictionary to store event instances (C# Programming Guide)
+
+One use for the `add` and `remove` custom event accessors is to expose many events without allocating a field for each event, but instead using a <xref:System.Collections.Generic.Dictionary%602> instance to store the event instances, as the example below demonstrates. This is only useful if a type has many events, but you expect that most of the events will not be subscribed to.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/events/how-to-use-a-dictionary-to-store-event-instances.md]: I suggest this addition:

> This implementation conforms to the behavior for [adding and removing delegates](../../../../_csharplang/spec/delegates#delegate-invocation) in the C# language specification.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Based on https://github.com/dotnet/samples/pull/701#issuecomment-472433057, I made the suggestion to link to the delegate addition and subtraction behavior in the C# language spec. After that, this is ready to merge.

File: docs/csharp/programming-guide/events/how-to-use-a-dictionary-to-store-event-instances.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/programming-guide/events/how-to-use-a-dictionary-to-store-event-instances.md]: @BillWagner I've added the suggested sentence. Please let me know, if it's not fine to have it after the example, as it's now.

**Review by BillWagner — APPROVED** [review]
Thanks for that final addition.  This LGTM @pkulikov 

I'll :shipit: now.

**Pull request merged**